src/Domain/Creators/SphereTimeDependentMaps.hpp

```diff
@@ -0,0 +1,185 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <memory>
+#include <optional>
+#include <string>
+#include <type_traits>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/Shape.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/SphericalCompression.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Frame {
+struct Grid;
+struct Distorted;
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace domain::creators::sphere {
+/*!
+ * \brief This holds all options related to the time dependent maps of the
+ * domain::creators::Sphere domain creator.
+ *
+ * \details Currently this class will only add a Size and a Shape map (and
+ * FunctionsOfTime) to the domain. Other maps can be added as needed.
+ *
+ * \note This struct contains no information about what blocks the time
+ * dependent maps will go in.
+ */
+struct TimeDependentMapOptions {
+ private:
+  template <typename SourceFrame, typename TargetFrame>
+  using MapType =
+      std::unique_ptr<domain::CoordinateMapBase<SourceFrame, TargetFrame, 3>>;
+  using IdentityMap = domain::CoordinateMaps::Identity<3>;
+  // Time-dependent maps
+  using CompressionMap =
+      domain::CoordinateMaps::TimeDependent::SphericalCompression<false>;
+  using ShapeMap = domain::CoordinateMaps::TimeDependent::Shape;
+
+  template <typename SourceFrame, typename TargetFrame>
+  using IdentityForComposition =
+      domain::CoordinateMap<SourceFrame, TargetFrame, IdentityMap>;
+  using GridToDistortedComposition =
+      domain::CoordinateMap<Frame::Grid, Frame::Distorted, ShapeMap,
+                            CompressionMap>;
+  using GridToInertialComposition =
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, ShapeMap,
+                            CompressionMap, IdentityMap>;
```

Comment:
I was under the assumption that since I (possibly) have a distorted -> inertial map which is just the identity, that I also need to include it in the overall grid -> inertial map so that they match. Is this not the case?

Reply:
you do not!